packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,835 @@
+                targets.length == signatures.length &&
+                targets.length == calldatas.length,
+            'NounsDAO::propose: proposal function information arity mismatch'
+        );
+        require(targets.length != 0, 'NounsDAO::propose: must provide actions');
+        require(targets.length <= proposalMaxOperations, 'NounsDAO::propose: too many actions');
+
+        temp.latestProposalId = latestProposalIds[msg.sender];
+        if (temp.latestProposalId != 0) {
+            ProposalState proposersLatestProposalState = state(temp.latestProposalId);
+            require(
+                proposersLatestProposalState != ProposalState.Active,
+                'NounsDAO::propose: one live proposal per proposer, found an already active proposal'
+            );
+            require(
+                proposersLatestProposalState != ProposalState.Pending,
+                'NounsDAO::propose: one live proposal per proposer, found an already pending proposal'
+            );
+        }
+
+        temp.startBlock = block.number + votingDelay;
+        temp.endBlock = temp.startBlock + votingPeriod;
+
+        proposalCount++;
+        Proposal storage newProposal = _proposals[proposalCount];
+        newProposal.id = proposalCount;
+        newProposal.proposer = msg.sender;
+        newProposal.proposalThreshold = temp.proposalThreshold;
+        newProposal.eta = 0;
+        newProposal.targets = targets;
+        newProposal.values = values;
+        newProposal.signatures = signatures;
+        newProposal.calldatas = calldatas;
+        newProposal.startBlock = temp.startBlock;
+        newProposal.endBlock = temp.endBlock;
+        newProposal.forVotes = 0;
+        newProposal.againstVotes = 0;
+        newProposal.abstainVotes = 0;
+        newProposal.canceled = false;
+        newProposal.executed = false;
+        newProposal.vetoed = false;
+        newProposal.totalSupply = temp.totalSupply;
+        newProposal.creationBlock = block.number;
+
+        latestProposalIds[newProposal.proposer] = newProposal.id;
+
+        /// @notice Maintains backwards compatibility with GovernorBravo events
+        emit ProposalCreated(
+            newProposal.id,
+            msg.sender,
+            targets,
+            values,
+            signatures,
+            calldatas,
+            newProposal.startBlock,
+            newProposal.endBlock,
+            description
+        );
+
+        /// @notice Updated event with `proposalThreshold` and `minQuorumVotes`
+        /// @notice `minQuorumVotes` is always zero since V2 introduces dynamic quorum with checkpoints
+        emit ProposalCreatedWithRequirements(
+            newProposal.id,
+            msg.sender,
+            targets,
+            values,
+            signatures,
+            calldatas,
+            newProposal.startBlock,
+            newProposal.endBlock,
+            newProposal.proposalThreshold,
+            newProposal.minQuorumVotes,
```

Comment:
Thinking more about this, seems beneficial to provide the quorum votes value at the time of proposal creation (`minQuorumVotes()`), rather than `0`.

Reply:
pushed a fix